src/librbd/crypto/CryptoObjectDispatch.cc

```diff
@@ -431,6 +432,8 @@ template <typename I>
 CryptoObjectDispatch<I>::CryptoObjectDispatch(
     I* image_ctx, CryptoInterface* crypto)
   : m_image_ctx(image_ctx), m_crypto(crypto) {
+  m_data_offset_object_no = Striper::get_num_objects(image_ctx->layout,
+                                                     crypto->get_data_offset());
```

Comment:
It should ;)  That case is not possible because `luks::FormatRequest` aligns the data offset (the boundary between CRYPTO_HEADER area and DATA area) to a stripe period and that is the only supported format handler right now.

Reply:
Ah, I saw it aligned it to a stripe period, but I confused the stripe period with the stripe unit. Ok.

Reply:
If an externally formatted image violates this requirement, `encryption_load` would fail -- see "librbd: check stripe pattern when loading encryption" commit in this PR.